orttraining/orttraining/python/ort_trainer.py

```diff
@@ -486,38 +486,65 @@ def __init__(self, model, loss_fn, model_desc, training_optimizer_name, map_opti
-                Inputs to the Combined PyTorch model are concatination of the 'model's input and 'loss_fn's label input.
-                Outputs of the Combined PyTorch model are concatination of 'loss_fn's loss output and 'model's outputs.
-            model_desc: model input and output description. it is used to specify input/output shapes, types, and names.
-                'model_desc' must be consistent with the training model.
-            training_optimizer_name: one of: 'SGDOptimizer', 'AdamOptimizer', 'LambOptimizer'.
-            map_optimizer_attributes: for optimizers with weight dependent parameters,
-                'map_optimizer_attributes' maps weight name to a set of optimization parameters.
-            learning_rate_description: in form of IODescription(Learning_Rate_Name, [1,], torch.float32).
-                Because learning_rate is an input to the training model, Learning_Rate_Name shall be set so that
-                there is no name conflict within the above model.
-            device: device to store tensors (e.g. 'cpu', 'cuda', 'cuda:<int_idx>')
-            gradient_accumulation_steps: number of training steps to accumulate gradients before averaging and applying them (default is 1)
-            postprocess_model: a callable to postprocess the ONNX model that is converted from PyTorch (default is None)
-            world_rank: rank id used for distributed training (default is 0)
-            world_size: number of ranks participating in distributed training (default is 1)
-            use_mixed_precision: flag to enable mixed precision (aka fp16) (default is False)
-            allreduce_post_accumulation: controls whether overlaping gradient computation with allreduce (default is False)
-            global_step: training step that will be used as input to 'get_lr_this_step'
-            get_lr_this_step: functor used as learning rate scheduler. It uses 'global_step' as input
-            loss_scaler: loss scaler that can be used used along with 'use_mixed_precision' to update loss scale automaticaly
-            partition_optimizer: controls whether to partition the optimizer state (default is False)
-            enable_grad_norm_clip: enables or disables gradient norm clipping
-            frozen_weights: list of model parameters to be frozen (not trained)
+        Initialize ORTTrainer.
+
+        :param model: one of
+               - a PyTorch model (class that inherits from torch.nn.Module)
+               - a combined PyTorch model and loss function.
+                  Inputs to this combined PyTorch model are a concatenation of the
+                  model's input and the loss function's label input.
+                  Outputs are a concatenation of the loss function's output and the
+                  model's output.
+               - a combined ONNX model and loss function.
+        :param loss_fn: one of
+               - a PyTorch loss function if 'model' is a PyTorch model. A loss
```

Comment:
worth mentioning that the loss function must be a `torch.nn.Module`

Reply:
Is that only when it is part of the combined model?

Reply:
loss_fn is either None (when it is present in the model) or torch.nn.Module, when it will be combined into the model and exported to ONNX. In fact, only torch.nn.Module can be exported into ONNX